README.md

```diff
@@ -0,0 +1,116 @@
+
+This software is a simulator to replicate the behavior of the Alternating Bit Protocol. The ABP simulator consists of 3 components.
+
+- **Receiver** - It receives the data and sends back acknowledgement after certain perios of time.
+- **Subnets** - It just transmits the packets to next node after a time delay. 
+- **Sender** - It sends the packet and then waits for acknowledgement from the receiver but if it does not receive the acknowledgement within time window it will resend the packet with alternating bit attached to it so that it can differentiate between the two packets. Once it receives the acknowledgement it will swnd the next packet until it has no more packets to send.
+### FILES ORGANIZATION
+---
+##### data [This folder contains the data files for the simulator]
+1. input
+    -   input_abp_0.txt
+    -    input_abp_1.txt
+2. output
+	-	-abp_output_0.txt
+	-	-abp_output_1.txt
+
+##### doc [This folder contains the documentation for the project]
+1. alternatebitprot.pdf
+2. Cadmium_Documentation_Ubuntu.pdf
+3. Cadmium_Windows.pdf
+
+##### include[This folder contains the header files]
+1. message.hpp
+2. receiver_cadmium.hpp
+3. sender_cadmium.hpp
+4. subnet_cadmium.hpp
+
+##### lib [This folder contains 3rd party libraries needed in the project]
+1. cadmium-master[This folder contains cadmium library files]
+2. DESTimes-master[This folder contains DesTimes library files]
+3. vendor[this folder contains third party vendor files]
+	-	iestream.hpp
+	-	nd_time.hpp
+
+##### src [This folder contains the source files written in c++ for the project]
+1. main.cpp
+2. message.cpp
+
+##### test [This folder the unit test for the different include files]
+1. data [This folder contains the data files for test folder]
+	1.1.receiver
+	-	receiver_input_test.txt
+	-	receiver_test_output.txt
+    
+    1.2.sender
+	-	sender_input_test_ack_In.txt
+	-   sender_input_test_control_In.txt
+	-   sender_test_output.txt
+		
+    1.3.subnet
+	-   subnet_input_test.txt
+	-   subnet_test_output.txt
+2. src [This folder contains the source files written in c++ for test folder]
+	2.1.receiver
+    -   main.cpp
+
+    2.2.sender
+	-   main.cpp
+	
+	2.3.subnet
+	-   main.cpp
+
+### STEPS TO RUN THE SIMULATOR
+---
+Document alternatebitprot.pdf in the doc folder contains the explanation of this simulator
+
+1.Update include paths in all the makefiles in this folder and subfolders. You need to update the following lines:
+
+1. INCLUDECADMIUM=-I lib/cadmium-master/include
+2. Update the relative path to cadmium/include from the folder where the makefile is. You need to take into account where you copied the folder during the installation process
+>               Example: INCLUDECADMIUM=-I lib/cadmium-master/include
+
```

Comment:
Before running any tests, the user would need to install any applicable dependencies and if the details are included in the README file, it will help user. For this purpose, you may provide steps to install dependencies or may guide to refer to the documentation section of the project.

Reply:
Solved